### DISCUSSION TITLE
Video thumbnail sometimes all black

### DISCUSSION THREAD

**Discussion by ChuckSavage** [dlemstra/Magick.NET#1167] [Help]
So, I'm using something relatively simple to get a thumbnail for a video file. At the moment, if a video uses fade in from black, the thumb is black. Is there a way to advance frame by frame to find a frame that isn't black?

```
using MagickImage image = new MagickImage(imageFile.FullName); 
image.Thumbnail(ThumbSizeDTO.Size.Width, ThumbSizeDTO.Size.Height);
image.Write(thumb.FullName);
```

I'm thinking before the `image.Write`, to resize the thumb to a 1x1 and check for blackness, if too black advance a frame and try again. Though I'm not sure how to step through the movie using MagickImage. I'm not seeing anything like `image.AdvanceFrame()` or some construct. The only Frame I see is for geometry stuff.

A `image.SeekFrame(frame_number)` method and a `image.FrameCount` property might help to create a for loop. Then a call to `image.Thumbnail` would use the current frame.

**Comment by dlemstra** [comment]
You will need to use an `MagickImageCollection` instead to iterate through the frames. But I would advise you to use the `MagickReadSettings` and set `FrameCount` to a value to make sure you only read the first `x` frames. And you could probably also use the `Histogram` method instead of resizing the image.
